src/Widgets/configuration/WidgetParametersConfiguration.tsx

```diff
@@ -0,0 +1,77 @@
+import { JSONSchema7, JSONSchema7Definition } from 'json-schema';
+import Form, { type ISubmitEvent, type UiSchema } from '@rjsf/core';
+import { useVerna } from '../../context/VernaContextProvider';
+import { FormEvent } from 'react';
+import WidgetParametersApplyer, { WidgetParameters } from './WidgetParametersApplyer';
+import { getTemplateWidgetName, SEPARATOR_ID_RJSF } from '../../utils';
+
+interface WidgetParametersConfigurationProps {
+  onClose: () => void;
+  widgetId: string;
+}
+
+export default function WidgetParametersConfiguration({
+  widgetId,
+  onClose,
+}: WidgetParametersConfigurationProps) {
+  const verna = useVerna();
+  const templateWidgetName = getTemplateWidgetName(widgetId, verna.uiSchema);
+
+  function handleSubmit(event: ISubmitEvent<unknown>, nativeEvent: FormEvent<HTMLFormElement>) {
+    nativeEvent.preventDefault();
+    WidgetParametersApplyer(event.formData as WidgetParameters, verna, widgetId);
+    onClose();
+    return false;
+  }
+
+  const uiSchema: UiSchema = {
+    SelectWidget: {
+      items: {
+        'ui:options': {
+          addable: true,
+          orderable: true,
+        },
+      },
+    },
+  };
+
+  function getDefaultParameterValues() {
+    const widgetIdParts = widgetId.split(SEPARATOR_ID_RJSF);
+    const currentWidgetName = widgetIdParts[verna.selector ? 1 : 2];
+    const currentSection = ((verna.selector
+      ? verna.schema
+      : verna.schema.properties?.[widgetIdParts[1]]) || {}) as {
+      [key: string]: JSONSchema7Definition;
+    };
+    const self =
+      ((verna.selector
+        ? currentSection?.[widgetIdParts[1]]
+        : (currentSection as JSONSchema7)?.properties?.[widgetIdParts[2]]) as JSONSchema7) || {};
```

Comment:
This seems weird... Line 45 you cast `currentSection` as `{[key:string]: JSONSchema7Definition}` then line 49, you cast the same variable as `JSONSchema7`. I think this hide something is wrong here.